app/screens/channel/channel_post_list/index.js

```diff
@@ -51,54 +48,4 @@ function mapDispatchToProps(dispatch) {
     };
 }
 
-function areStatesEqual(next, prev) {
-    const nextChannelId = next.entities.channels.currentChannelId;
-    const prevChannelId = prev.entities.channels.currentChannelId;
-    if (prevChannelId !== nextChannelId) {
-        return false;
-    }
-    const nextVisible = next.views.channel.postVisibility[nextChannelId];
-    const prevVisible = prev.views.channel.postVisibility[nextChannelId];
-    const nextPostsIds = next.entities.posts.postsInChannel[nextChannelId];
-    const prevPostsIds = prev.entities.posts.postsInChannel[nextChannelId];
-
-    // When we don't have posts and we failed to get the post after tha max retry attempts
-    const prevRetryFailed = prev.views.channel.retryFailed;
-    const nextRetryFailed = next.views.channel.retryFailed;
-    if (prevRetryFailed !== nextRetryFailed && nextRetryFailed && !nextPostsIds) {
-        return false;
-    }
-
-    // if we don't have post and visibility is not set we don't need to re-render
-    if (!nextPostsIds || !nextVisible || !prevVisible) {
-        return true;
-    }
-
-    const nextVisiblePostsIds = nextPostsIds.slice(0, nextVisible);
-    const prevVisiblePostsIds = prevPostsIds ? prevPostsIds.slice(0, prevVisible) : [];
-
-    // if we have a different amount of post we should re-render
-    if (nextVisiblePostsIds.length !== prevVisiblePostsIds.length) {
-        return false;
-    }
-
-    const {status: nextStatus} = next.requests.posts.getPosts;
-    const {status: prevStatus} = prev.requests.posts.getPosts;
-
-    // if we are requesting post for the first time we should re-render
-    if (prevStatus === RequestStatus.STARTED && nextStatus === RequestStatus.SUCCESS &&
-        nextPostsIds.length !== prevPostsIds.length) {
-        return false;
-    }
-
-    // if at least one post id changed we need to re-render
-    for (let i = 0; i <= nextVisiblePostsIds.length; i++) {
-        if (nextVisiblePostsIds[i] && nextVisiblePostsIds[i] !== prevVisiblePostsIds[i]) {
-            return false;
-        }
-    }
-
-    return true;
-}
-
-export default connect(makeMapStateToProps, mapDispatchToProps, null, {pure: true, areStatesEqual})(ChannelPostList);
+export default connect(makeMapStateToProps, mapDispatchToProps)(ChannelPostList);
```

Comment:
nothing to do here, I think this by using this selector makes it easier to understand, I really like it cause the selector is not expensive at all